src/tests/ftest/osa/osa_offline_drain.py

```diff
@@ -69,45 +87,143 @@ def run_offline_drain_test(self, num_pool, data=False,
-            self.log.info("Pool Version at the beginning %s", pver_begin)
-            if self.test_during_aggregation is True:
-                self.pool.set_property("reclaim", "time")
-                self.delete_extra_container(self.pool)
-                self.simple_exclude_reintegrate_loop(rank)
-            output = self.dmg_command.pool_drain(self.pool.uuid,
-                                                 rank, t_string)
-            self.print_and_assert_on_rebuild_failure(output)
-
-            pver_drain = self.get_pool_version()
-            self.log.info("Pool Version after drain %d", pver_drain)
-            # Check pool version incremented after pool drain
-            self.assertTrue(pver_drain > (pver_begin + 1),
-                            "Pool Version Error:  After drain")
+            # Drain ranks provided in YAML file
+            for index, rank in enumerate(self.ranks):
+                self.pool = pool[val]
+                # If we are testing using multiple pools, reintegrate
+                # the rank back and then drain.
+                self.pool.display_pool_daos_space("Pool space: Beginning")
+                pver_begin = self.get_pool_version()
+                self.log.info("Pool Version at the beginning %s", pver_begin)
+                if self.test_during_aggregation is True and index == 0:
+                    self.pool.set_property("reclaim", "time")
+                    self.delete_extra_container(self.pool)
+                    self.simple_drain_reintegrate_loop(rank)
+                if (self.test_during_rebuild is True and val == 0):
+                    # Exclude rank 3
+                    output = self.dmg_command.pool_exclude(self.pool.uuid, "3")
+                    self.pool.wait_for_rebuild(True)
+                output = self.dmg_command.pool_drain(self.pool.uuid,
+                                                     rank, t_string)
+                self.print_and_assert_on_rebuild_failure(output)
+
+                pver_drain = self.get_pool_version()
+                self.log.info("Pool Version after drain %d", pver_drain)
+                # Check pool version incremented after pool drain
+                self.assertTrue(pver_drain > (pver_begin + 1),
+                                "Pool Version Error:  After drain")
+                if num_pool > 1:
+                    output = self.dmg_command.pool_reintegrate(self.pool.uuid,
+                                                               rank, t_string)
+                    self.print_and_assert_on_rebuild_failure(output)
+                if (self.test_during_rebuild is True and val == 0):
+                    # Reintegrate rank 3
+                    output = self.dmg_command.pool_reintegrate(self.pool.uuid,
+                                                               "3")
+                    self.print_and_assert_on_rebuild_failure(output)
 
         for val in range(0, num_pool):
```

Comment:
Minor: If rebasing, it will be nice to have a comment for this block explaining what it is doing.

Reply:
Looks like all tests passed again... Don't want to rebase... Created DAOS-7231 to perform the changes requested.